### PR TITLE
Implement task cloning in experiments table [SCI-7382]

### PR DESCRIPTION
**Pull request by artoscinote**
Jira ticket: [SCI-7382](https://scinote.atlassian.net/browse/SCI-7382)

### What was done
Implement task cloning in experiments table

### CODE REVIEW

File: app/assets/javascripts/experiments/table.js [hunk 1]
```diff
@@ -285,15 +306,51 @@ var ExperimnetTable = {
           return { ...params, ...{ filters: this.activeFilters } };
         }
       });
+
+      this.initProvisioningStatusPolling();
     });
   },
+  initProvisioningStatusPolling: function() {
+    let provisioningStatusUrls = $('[data-urls]').toArray()
```
Comment (aignatov-bio) [inline app/assets/javascripts/experiments/table.js:314]: Maybe get provisioning rows and only URLs for this modules - `this.getUrls().provisioning_status

File: app/assets/javascripts/experiments/table.js [hunk 2]
```diff
@@ -285,15 +306,51 @@ var ExperimnetTable = {
...
+          if (data.provisioning_status === 'in_progress') remainingUrls.push(url);
+        },
+        async: false
+      });
+    });
+
+    if (remainingUrls.length > 0) {
+      setTimeout(() => {
+        this.pollProvisioningStatuses(remainingUrls);
+      }, 5000);
```
Comment (aignatov-bio) [inline app/assets/javascripts/experiments/table.js:337]: Maybe more 10 seconds?

File: app/models/concerns/cloneable.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module Cloneable
```
Comment (aignatov-bio) [inline app/models/concerns/cloneable.rb:3]: Did you check this helper - `module RenamingUtil`

File: app/assets/javascripts/experiments/table.js [hunk 4]
```diff
@@ -21,14 +21,26 @@ var ExperimnetTable = {
...
-          <div class="sci-checkbox-container">
-            <input type="checkbox" class="sci-checkbox my-module-selector" data-my-module="${id}">
-            <span class="sci-checkbox-label"></span>
-          </div>
-        </div>`;
+      let row;
+      let provisioningStatus = data.provisioning_status;
+
+      // Checkbox selector or provisioning spinner
+      if (provisioningStatus === 'in_progress') {
```
Comment (aignatov-bio) [inline app/assets/javascripts/experiments/table.js:28]: Can we just use class to row during provisioning to hide/show required information?

**Review by aignatov-bio — CHANGES_REQUESTED** [review]

**Review by aignatov-bio — COMMENTED** [review]
Do we need update my_module permissions, to prevent mange during cloning?

File: app/models/concerns/cloneable.rb (hunk 3 above)
Comment (artoscinote) [reply app/models/concerns/cloneable.rb:3]: That doesn't work as specified (specification is it should work the same as experiments). Also, I think that code should be either refactored, or scrapped.

File: app/models/concerns/cloneable.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  extend ActiveSupport::Concern
+
+  def next_clone_name
+    raise NotImplementedError, "Cloneable model must implement the '.parent' method!" unless respond_to?(:parent)
+
+    last_clone_number =
+      parent.public_send(self.class.table_name)
+            .select("substring(#{self.class.table_name}.name, '(?:^Clone )(\\d+)')::int AS clone_number")
```
Comment (okriuchykhin) [inline app/models/concerns/cloneable.rb:11]: Maybe it makes sense to move "Clone" to I18N?

**Review by aignatov-bio — APPROVED** [review]

**Pull request merged**